iocore/net/SSLSNIConfig.cc

```diff
@@ -166,13 +167,19 @@ SNIConfigParams::Initialize()
     return 1;
   }
 
-  ts::Errata zret = Y_sni.loader(sni_filename);
+  YamlSNIConfig Y_sni_tmp;
+  ts::Errata zret = Y_sni_tmp.loader(sni_filename);
   if (!zret.isOK()) {
     std::stringstream errMsg;
     errMsg << zret;
-    Error("%s failed to load: %s", sni_filename, errMsg.str().c_str());
+    if (TSSystemState::is_initializing()) {
+      Emergency("%s failed to load: %s", sni_filename, errMsg.str().c_str());
+    } else {
+      Error("%s failed to load: %s", sni_filename, errMsg.str().c_str());
+    }
     return 1;
   }
+  Y_sni = std::move(Y_sni_tmp);
```

Comment:
I added this logic to make sure the static data remained the coherent, old version if an invalid sni.yaml was reloaded.